src/main/java/org/broadinstitute/hellbender/engine/datasources/ReadsHeaderSource.java

```diff
@@ -0,0 +1,75 @@
+package org.broadinstitute.hellbender.engine.datasources;
+
+import com.google.api.services.storage.Storage;
+import com.google.cloud.genomics.dataflow.readers.bam.BAMIO;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SamReader;
+import htsjdk.samtools.SamReaderFactory;
+import htsjdk.samtools.ValidationStringency;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.broadinstitute.hellbender.engine.spark.AuthHolder;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.seqdoop.hadoop_bam.util.SAMHeaderReader;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+/**
+ * Class to read BAM headers. May be expanded to include BAM contents later, if useful.
```

Comment:
Our existing `ReadsSource` classes already have `getHeader() methods, so it's likely we will eventually refactor this class away.